arch/riscv/core/isr.S

```diff
@@ -66,6 +69,156 @@
+	RV_OP_STOREREG temp, __z_arch_esf_t_fa4_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_fa5_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_fa5_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_fa6_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_fa6_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_fa7_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_fa7_OFFSET(to_reg)		;
+
+#define COPY_ESF(to_reg, from_reg, temp)				\
+	RV_OP_LOADREG temp, __z_arch_esf_t_mepc_OFFSET(from_reg)	;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_mepc_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_mstatus_OFFSET(from_reg)	;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_mstatus_OFFSET(to_reg)	;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_ra_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_ra_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_gp_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_gp_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_tp_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_tp_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_t0_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_t0_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_t1_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_t1_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_t2_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_t2_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_t3_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_t3_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_t4_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_t4_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_t5_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_t5_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_t6_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_t6_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_a0_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_a0_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_a1_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_a1_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_a2_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_a2_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_a3_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_a3_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_a4_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_a4_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_a5_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_a5_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_a6_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_a6_OFFSET(to_reg)		;\
+	RV_OP_LOADREG temp, __z_arch_esf_t_a7_OFFSET(from_reg)		;\
+	RV_OP_STOREREG temp, __z_arch_esf_t_a7_OFFSET(to_reg)		;
+
+#define DO_CALLER_SAVED(op, reg) \
```

Comment:
The naming of CALLER and CALLEE macros are reversed.

Reply:
Right, I will fix it